internal/chart/helm_client.go

```diff
@@ -2,23 +2,33 @@ package chart
 
 import (
 	"errors"
+	"fmt"
 	"log"
 	"os"
+	"time"
 
+	"github.com/go-logr/logr"
 	"helm.sh/helm/v3/pkg/action"
 	"helm.sh/helm/v3/pkg/chart/loader"
 	"helm.sh/helm/v3/pkg/release"
 	"helm.sh/helm/v3/pkg/storage/driver"
+	helmTime "helm.sh/helm/v3/pkg/time"
+
 	"k8s.io/cli-runtime/pkg/genericclioptions"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+const (
+	defaultDeploymentTimeout = 10 * time.Minute
```

Comment:
usually, it takes like 2-3 seconds to render a helm chart and deploy it.
How about a one-minute timeout?